recipes-bsp/u-boot/files/0028-board-siemens-iot2050-implement-the-load-and-configu.patch

```diff
@@ -0,0 +1,342 @@
++				serdes_mux_ctrl_info.ctrl_pcie1_pcie0);
++	set_pinvalue(serdes_mux_ctl_pin_info[2].gpio_name, serdes_mux_ctl_pin_info[2].label,
++				serdes_mux_ctrl_info.ctrl_usb30_pcie0_lane1);
++}
++
++
++static void set_m2_load_info(const char **fdt_name)
++{
++	int i;
++	int config_table_size = ARRAY_SIZE(m2_config_table);
++	struct m2_device_config_decodes	config;
++	struct serdes_combination current_serdes_combination =
++												{BKEY_USB30_EKEY_PCIE, USB30, PCIE1_LANE0};
++	enum m2_connector_combination_type current_m2_connector_type;
++
++	if (env_get("m2_manual_config")) {
++		printf("Maunal Select M.2 Configuration\n");
++		current_m2_connector_type = simple_strtoul(env_get("m2_manual_config"), NULL, 10);
++		get_serdes_combination(&current_serdes_combination, current_m2_connector_type);
++	} else { /*auto detect*/
++		get_m2_config_decodes(&config);
++		for (i = 0; i < config_table_size; i++) {
++			if (!memcmp(&config, &m2_config_table[i].m2_config_decodes,
++						sizeof(struct m2_device_config_decodes))) {
++				current_serdes_combination =  m2_config_table[i].serdes_select;
++				break;
++			}
++		}
++	}
++
++	update_m2_load_info(current_serdes_combination, fdt_name);
++	configure_serdes_mux(current_serdes_combination);
++}
++
++/*
++ * M.2 Board is a Variant Baesed On PG2 board
++ */
++static bool board_is_m2(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++
++	return info->magic == IOT2050_INFO_MAGIC &&
++		strcmp((char *)info->name, "IOT2050-ADVANCED-M2") == 0;
++}
++
++static void m2_power_enable(void)
++{
++	/* enable m.2 connector power */
++	set_pinvalue(m2_enable_power.gpio_name, m2_enable_power.label, 1);
++	udelay(4 * 100);
++}
++
+ static bool board_is_advanced(void)
+ {
+ 	struct iot2050_info *info = IOT2050_INFO_DATA;
+@@ -79,7 +318,7 @@ void set_board_info_env(void)
+ 		return;
+ 	}
+ 
+-	if (env_get("board_uuid"))
++	if (env_get("board_uuid") && !board_is_m2())
+ 		return;
+ 
+ 	env_set("board_name", info->name);
+@@ -89,7 +328,6 @@ void set_board_info_env(void)
+ 	env_set("board_a5e", info->a5e);
+ 	env_set("fw_version", PLAIN_VERSION);
+ 	env_set("seboot_version", info->seboot_version);
+-
+ 	if (IS_ENABLED(CONFIG_NET)) {
+ 		/* set MAC addresses to ensure forwarding to the OS */
+ 		for (mac_cnt = 0; mac_cnt < info->mac_addr_cnt; mac_cnt++) {
+@@ -103,6 +341,10 @@ void set_board_info_env(void)
+ 	if (board_is_advanced()) {
+ 		if (IS_ENABLED(CONFIG_TARGET_IOT2050_A53_PG1))
+ 			fdtfile = "ti/k3-am6548-iot2050-advanced.dtb";
++		else if (board_is_m2()) {
++			m2_power_enable();
++			set_m2_load_info(&fdtfile);
++		}
+ 		else
+ 			fdtfile = "ti/k3-am6548-iot2050-advanced-pg2.dtb";
+ 	} else {
+@@ -225,6 +467,17 @@ int board_late_init(void)
+ 	return 0;
+ }
+ 
++static void m2_fdt_fixup(void *blob)
++{
++	int ret;
++
++	ret = fdt_find_and_setprop(blob, "/", "compatible", fdt_fix_string,
++								strlen(fdt_fix_string) + 1, 0);
++
```

Comment:
unneeded newline

Reply:
removed the unneeded line.close the conversation